File: 프로그래머스/unrated/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.js
function solution(arr) {
    let times = 0;
    
    while (true) {
        if (2 ** times >= arr.length) {
            break;
        }
        
        times++;
    }
    
    if (2 ** times === arr.length) {
        return arr;
    }
    
    times = 2 ** times - arr.length;
    
    for (let i = 0; i < times; i++) {
        arr.push(0);
    }
    
    return arr;
}

File: 프로그래머스/unrated/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.java
class Solution {
    public int[] solution(int[] arr) {
        int pow = 0;
        while (true) {
            if (Math.pow(2, pow) >= arr.length) {
                break;
            }
            
            pow++;
        }
        
        if (Math.pow(2, pow) == arr.length) {
            return arr;
        }
        
        int[] result = new int[(int) Math.pow(2, pow)];
        for (int i = 0; i < arr.length; i++) {
            result[i] = arr[i];
        }
        
        return result;
    }
}